messaging/src/test/java/org/axonframework/commandhandling/AsynchronousCommandBusTest.java

```diff
@@ -68,9 +71,15 @@ void setUp() throws Exception {
     void dispatchWithCallback() throws Exception {
         testSubject.subscribe(Object.class.getName(), commandHandler);
         CommandCallback<Object, Object> mockCallback = mock(CommandCallback.class);
+        when(mockCallback.wrap(any())).thenCallRealMethod();
         CommandMessage<Object> command = asCommandMessage(new Object());
         testSubject.dispatch(command, mockCallback);
 
+        spanFactory.verifySpanCompleted("SimpleCommandBus.dispatch");
```

Comment:
Exactly this may be off to end users when combining this tracing logic with the async `CommandBus`.

Reply:
Yes you are correct, adjusting the tests after changing the bus implementation